saleor/product/tasks.py

```diff
@@ -31,3 +38,30 @@ def update_variants_names(product_type_pk, saved_attributes_ids):
     instance = ProductType.objects.get(pk=product_type_pk)
     saved_attributes = Attribute.objects.filter(pk__in=saved_attributes_ids)
     return _update_variants_names(instance, saved_attributes)
+
+
+@app.task
+def update_product_minimal_variant_price_task(product_pk):
+    product = Product.objects.get(pk=product_pk)
+    update_product_minimal_variant_price(product)
+
+
+@app.task
+def update_products_minimal_variant_prices_of_catalogues_task(
+    product_ids=[], category_ids=[], collection_ids=[]
+):
+    update_products_minimal_variant_prices_of_catalogues(
+        product_ids, category_ids, collection_ids
+    )
+
+
+@app.task
+def update_products_minimal_variant_prices_of_discount_task(discount_pk):
+    discount = Sale.objects.get(pk=discount_pk)
+    update_products_minimal_variant_prices_of_discount(discount)
+
+
+@app.task
+def update_all_products_minimal_variant_prices_task():
```

Comment:
Do we need this task? All products will always have the up to date `minimal_variant_prices`. So there is no need to have this task. Am I right?

Reply:
I thought so myself but then I've recalled that sales can expire. Even worse they can now expire or start and any hour of the day!

We need to establish default way for saleor to spawn periodical tasks, then we can track what Sales' start/end in the refresh window and recalculate affected product's minimal prices. However, I would still run this `update all` task once ~at midnight just to be sure.